cloud-elements/churros
# Element extend tests

## Highlights
* Get a system element, count the resources
* Tests for delete new/override resources
* Tests for creating a resource under system account
* Get all resources to see if the required resource is present
* Tests for get accountOnly resources to see if the required resource is present
* Tests for get the newly created resource
* Tests for accessing the resources by creating a new user under different account
* Execute the newly created resource and Execute the newly created resource and system resource
* Tests for Override a system resource
* Execute the overriden created resource
* Tests for adding/updating parameter to system resource
* Tests for deleting system resource parameter
* Tests for adding/update parameter to new resource
* Tests for delete new resource parameter
* Tests for add/delete model to system resource and should fail
* Tests for add/delete model to new resource
* Tests for creating/deleteing/executing transformations for newly created model
* Tests for add/delete model to new resource

## Examples
```
  element-extend
    ✓ should create new resource under system catalog element (194ms)
    ✓ should get system catalog element with resources and account resources (200ms)
    ✓ should get all resources of system catalog element and account resources (344ms)
    ✓ should get accountOnly resources (198ms)
    ✓ should get the newly created resource for the element (214ms)
    ✓ should create a new account and element in new account should not have access to new account resources (869ms)
    ✓ should support creating an instance and execute the newly created instance (1868ms)
    ✓ should override the existing system catalog resource as new resource for the element (199ms)
    ✓ should get accountOnly resources (209ms)
    ✓ should support creating an instance and execute the newly created instance (1032ms)
    ✓ should return error while adding/updating or deleting parameters for system catalog resource (533ms)
    ✓ should CUD a parameter for newly created resource (1340ms)
    ✓ should return error while adding or deleting resource model for system catalog resource (367ms)
    ✓ should be create and delete a model for newly created resource (364ms)
    ✓ should be create/execute and delete transformation on the newly created model for newly created resource (10885ms)
    ✓ should get system catalog element with resources and no account resources (251ms)
```
